tests/modal.ts

```diff
@@ -0,0 +1,1097 @@
+        `);
+        const modal = el1.querySelector("#modal");
+        modal.animationDur = 0;
+        const slot_id = el1.querySelector("#slot");
+        const modal_par = modal.parentElement;
+        modal.show();
+        const new_btn = modal.querySelector("#new_button");
+        const new_el=document.createElement("button");
+        new_el.classList.add("closing");
+        slot_id.appendChild(new_el);
+        const new_el_wrap = new_el.getBoundingClientRect();
+        expect(modal.parentElement).to.equal(document.body);
+        new_el.click();
+        expect(modal.parentElement).to.equal(modal_par);
+    });
+
+    it('noesc', async () => {
+        const el1 = await fixture(html`
+            <div id="parent">
+                <aalam-modal id="modal">
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div>
+        `);
+        const modal = el1.querySelector("#modal");
+        const modal_parent = modal.parentElement;
+        modal.show();
+        modal.animationDur = 0;
+        expect(modal.parentElement).to.equal(document.body);
+        await sendKeys({down:"Escape"});
+        expect(modal.parentElement).to.equal(modal_parent);
+
+        const el2 = await fixture(html `
+            <div id="parent">
+                <aalam-modal id="modal" noesc>
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div> `);
+        const el_modal = el2.querySelector("#modal");
+        el_modal.show();
+        el_modal.animationDur = 0;
+        expect(el_modal.parentElement).to.equal(document.body);
+        await sendKeys({down:"Escape"});
+        expect(el_modal.parentElement).to.equal(document.body);
+    });
+
+    it('bgclose', async () => {
+        const el1 = await fixture(html`<div>
+            <div id="parent">
+                <aalam-modal id="modal">
+                        <div id="modal_body" slot="modal-body">
+                            <h2 id="c_click">h2 </h2>
+                        </div>
+                </aalam-modal>
+            </div></div>
+            `);
+        const modal = el1.querySelector("#modal");
+        await modal.updated();
+        modal.animationDur=0;
+        const modal_body = el1.querySelector("#modal_body");
+        const modal_par = modal.parentElement;
+        modal.show();
+        await modal.updated()
+        expect(modal.parentElement).to.equal(document.body);
+        const c_click = modal.querySelector("#c_click");
+        c_click.click();
+        expect(modal.parentElement).to.equal(document.body);
+        const wrapper_bounds = modal_body.getBoundingClientRect();
+
+        await sendMouse({type:'move', position: [wrapper_bounds.width - 200,
+                         Math.round(wrapper_bounds.y - 40)]});
+        await modal.updated();
+        await sendMouse({type:'down'});
+        await sendMouse({type:'up'});
+        expect(modal.parentElement).to.equal(modal_par);
```

Comment:
Have to test with nobgclose attribute. You can use the same element and setAttribute and then see if the click isnt closing the modal.